### ISSUE TITLE
Implement scaffolding for an example data loader

### ISSUE THREAD

**Issue by drewoldag** [lincc-frameworks/fibad#13]
Roughly speaking we can use the following guides as a starting point: 

1. https://pytorch.org/tutorials/beginner/blitz/cifar10_tutorial.html#load-and-normalize-cifar10
2. https://pytorch.org/tutorials/beginner/basics/data_tutorial.html#creating-a-custom-dataset-for-your-files
3. https://www.tensorflow.org/tutorials/images/cnn#download_and_prepare_the_cifar10_dataset

All in all, the machinery will likely need to be similar to what was implemented for the dynamic model loading, namely the user should be able to specify either a built in data loader or and the various parameters needed, or specify an external data loader that can be dynamically imported and used.
